patches/server/0575-Configurable-door-breaking-difficulty.patch

```diff
@@ -5,58 +5,66 @@ Subject: [PATCH] Configurable door breaking difficulty
-+            "door-breaking-difficulty.zombie",
-+            java.util.Arrays.stream(net.minecraft.world.Difficulty.values())
-+                .filter(net.minecraft.world.entity.monster.Zombie.DOOR_BREAKING_PREDICATE)
-+                .collect(Collectors.toList()),
-+            net.minecraft.world.Difficulty.class
-+        );
-+        vindicatorBreakDoors = getEnumList(
-+            "door-breaking-difficulty.vindicator",
-+            java.util.Arrays.stream(net.minecraft.world.Difficulty.values())
-+                .filter(net.minecraft.world.entity.monster.Vindicator.DOOR_BREAKING_PREDICATE)
-+                .collect(Collectors.toList()),
-+            net.minecraft.world.Difficulty.class
-+        );
++        for (final net.minecraft.world.entity.EntityType<?> entityType : entitiesWhoCanBreakDoors) {
++            java.util.function.Predicate<net.minecraft.world.Difficulty> difficultyPredicate = net.minecraft.world.entity.monster.Zombie.DOOR_BREAKING_PREDICATE;
++            if (entityType == net.minecraft.world.entity.EntityType.VINDICATOR) {
++                difficultyPredicate = net.minecraft.world.entity.monster.Vindicator.DOOR_BREAKING_PREDICATE;
++            }
++            entitiesDifficultyBreakDoors.put(
++                entityType,
++                getEnumList(
++                    "door-breaking-difficulty.".concat(entityType.id),
++                    java.util.Arrays.stream(net.minecraft.world.Difficulty.values())
++                        .filter(difficultyPredicate)
++                        .collect(Collectors.toList()),
++                    net.minecraft.world.Difficulty.class
++                )
++            );
++        }
 +    }
 +
      public short keepLoadedRange;
      private void keepLoadedRange() {
          keepLoadedRange = (short) (getInt("keep-spawn-loaded-range", Math.min(spigotConfig.viewDistance, 10)) * 16);
 diff --git a/src/main/java/net/minecraft/world/entity/monster/Vindicator.java b/src/main/java/net/minecraft/world/entity/monster/Vindicator.java
-index ddf77350b1762be523eeed760669bc33f634109f..85294ab8eaa654be9358f489cec7e1ae665fc34b 100644
+index ddf77350b1762be523eeed760669bc33f634109f..f51868f7b42b58901ff1c0ff1427d8071bfbe530 100644
 --- a/src/main/java/net/minecraft/world/entity/monster/Vindicator.java
 +++ b/src/main/java/net/minecraft/world/entity/monster/Vindicator.java
 @@ -195,7 +195,7 @@ public class Vindicator extends AbstractIllager {
  
      static class VindicatorBreakDoorGoal extends BreakDoorGoal {
          public VindicatorBreakDoorGoal(Mob mob) {
 -            super(mob, 6, Vindicator.DOOR_BREAKING_PREDICATE);
-+            super(mob, 6, com.google.common.base.Predicates.in(mob.level.paperConfig.vindicatorBreakDoors)); // Paper
++            super(mob, 6, com.google.common.base.Predicates.in(mob.level.paperConfig.entitiesDifficultyBreakDoors.getOrDefault(mob.getType(), mob.level.paperConfig.entitiesDifficultyBreakDoors.get(EntityType.VINDICATOR)))); // Paper
```

Comment:
Replaced again the logic here.. i prefer base the get difficulty in the mob using in the constructor... if cant get then use the entity for what this clase was made... only in the strange case a user with NMS try use this class... not sure can happen but better make this case (?